intranethome/templates/intranethome/mail_aliases.html

```diff
@@ -0,0 +1,178 @@
+<a href="/mailaliases/q/">Q</a> &nbsp; 
+<a href="/mailaliases/r/">R</a> &nbsp; 
+<a href="/mailaliases/s/">S</a> &nbsp; 
+<a href="/mailaliases/t/">T</a> &nbsp; 
+<a href="/mailaliases/u/">U</a> &nbsp; 
+<a href="/mailaliases/v/">V</a> &nbsp; 
+<a href="/mailaliases/w/">W</a> &nbsp; 
+<a href="/mailaliases/x/">X</a> &nbsp; 
+<a href="/mailaliases/y/">Y</a> &nbsp; 
+<a href="/mailaliases/z/">Z</a> &nbsp; 
+<a href="/mailaliases/number/">#</a>
+</h3>
+
+    {%for key, value in cleaned_data.items%}
+
+    <button class="accordion">{{key}}</button>
+    <div class="panel">
+      <table class="table">
+	<br>
+        {%for note_email in value%}
+          {%if note_email.note%}
+        	<tbody>
+		  <td> note: {{note_email.note}} </td>
+		</tbody>
+
+          {%elif note_email.triangle_brackets%}
+		  <tbody>
+                    <td><a href="mailto:{{note_email.triangle_brackets.1}}">{{note_email.triangle_brackets.0}}</a></td>
+                    <td>{{note_email.triangle_brackets.2}}</td>
+		  </tbody>
+
+          {%elif note_email.parentheses%}
+		  <tbody>
+                    <td><a href="mailto:{{note_email.parentheses.1}}">{{note_email.parentheses.0}}</a></td>
+                    <td>{{note_email.parentheses.2}}</td>
+		  </tbody>
+         
+	  {%elif note_email.plain_email%}
+		<tbody>
+		    <td><a href="mailto:{{note_email.plain_email}}">{{note_email.plain_email}}</a></td>
+		</tbody>
+            
+          {%else%}
+		<tbody>
+		  <td>{{note_email.local}}</td>
+		</tbody>
+          {%endif%}
+
+        {%endfor%}
+	</table>
+      	<br>
+    </div>
+    {%endfor%}
+{%endif%}
+
+<script>
+    
+    function singleAccordionToggle() {
+        this.classList.toggle("active");
+        var panel = this.nextElementSibling;
+        if (panel.style.maxHeight) {
+          panel.style.maxHeight = null;
+        } else {
+          panel.style.maxHeight = 0;
+        } 
+      }
+      
+    function enableButtonListening(){
+      const accordionElements = document.getElementsByClassName("accordion");
+      var i;
+    
+      for (i = 0; i < accordionElements.length; i++) {
+        accordionElements[i].addEventListener("click", singleAccordionToggle);
+      }
+    }
+
+    function expandAll(){
+      const accordionElements = document.getElementsByClassName("accordion");
+      var i;
+
+      for (i = 0; i < accordionElements.length; i++) {
+        accordionElements[i].classList.toggle("active");
+        var panel = accordionElements[i].nextElementSibling;
+        panel.style.maxHeight = null;
+      }
+    }
+
+    function contractAll(){
+      const accordionElements = document.getElementsByClassName("accordion");
+      var i;
+
+      for (i = 0; i < accordionElements.length; i++) {
+        accordionElements[i].classList.toggle("active");
+        var panel = accordionElements[i].nextElementSibling;
+        panel.style.maxHeight = 0;
+      }
+    }
+
+    enableButtonListening();
+    contractAll();
+
+    </script>
```

Comment:
Nice Javascript! Very clean and easy to understand.

Reply:
Seconded!  The original example code was kinda gross.